gulpfile.js

```diff
@@ -48,85 +52,98 @@ function formatByteMessage (source, data) {
   return chalk`{cyan ${source.padStart(12, ' ')}}: {bold ${data.fileName}} ${message}`
 }
 
-function style () {
-  const isLegacy = path => /legacy/.test(path)
+const style = () => {
+  // Don't inline minified versions, so builds can lazily import them at runtime
+  const cssImportOptions = { filter: (path) => !/\.min/.test(path) }
 
-  const excludeModern = filter(file => isLegacy(file.path), { restore: true })
-  const excludeLegacy = filter(file => !isLegacy(file.path), { restore: true })
+  const startDiff = () => bytediff.start()
+  const endDiff = (source) => bytediff.stop((data) => formatByteMessage(source, data))
 
   return (
     gulp
       .src(paths.styles.src)
-      // Add sourcemaps
       .pipe(sourcemaps.init())
-      // Resolve imports, calculated colors and inlined SVG files
-      .pipe(postcss([postcssImport(), postcssColorModFunction(), postcssInlineSvg()]))
-
-      // * Process legacy builds *
-      .pipe(excludeModern)
-      // Inline variable values so CSS works in legacy browsers
-      .pipe(postcss([postcssCssVariables()]))
-      // Calculate size before autoprefixing
-      .pipe(bytediff.start())
-      // autoprefix
-      .pipe(postcss([autoprefixer({
-        env: 'legacy'
-      })]))
-      // Write the amount gained by autoprefixing
-      .pipe(bytediff.stop(data => formatByteMessage('autoprefixer', data)))
-      .pipe(excludeModern.restore)
-
-      // * Process modern builds *
-      .pipe(excludeLegacy)
-      // Calculate size before autoprefixing
-      .pipe(bytediff.start())
-      // autoprefix modern builds
-      .pipe(postcss([autoprefixer({
-        env: 'modern'
-      })]))
-      // Write the amount gained by autoprefixing
-      .pipe(bytediff.stop(data => formatByteMessage('autoprefixer', data)))
-      .pipe(excludeLegacy.restore)
-
-      // Write the sourcemaps after making pre-minified changes
+      .pipe(postcss([postcssImport(cssImportOptions), postcssColorModFunction(), postcssInlineSvg()]))
+
+      .pipe(startDiff())
+      .pipe(postcss([postcssCssVariables({ preserve: true })]))
+      .pipe(endDiff('css variables'))
+
+      .pipe(startDiff())
+      .pipe(postcss([autoprefixer({ env: 'legacy' })]))
```

Comment:
We don't have `env: 'modern'` anymore, so we could simplify this to one default "env-less" browserslist and remove the options object here.

Reply:
+1

Reply:
Moving to #197 :)